docs/developer-docs/latest/setup-deployment-guides/deployment/hosting-guides/heroku.md

```diff
@@ -287,116 +110,187 @@ module.exports = ({ env }) => ({
 
-<code-group>
+### Install and use the Heroku CLI
+
+1. Use the following OS-specific installation instructions to install the Heroku CLI tool:
+
+      :::: tabs card
+
+      ::: tab Ubuntu
+      Run the following from your terminal:
+
+      ```bash
+      sudo snap install --classic heroku
+      ```
+
+      :::
+
+      ::: tab Mac
+      [Download the installer](https://cli-assets.heroku.com/heroku.pkg)
+
+      Also available via Homebrew:
+
+      ```bash
+      brew tap heroku/brew && brew install heroku
+      ```
+
+      :::
+
+      ::: tab Windows
+      Download the appropriate installer for your Windows installation:
+
+      - [64-bit installer](https://cli-assets.heroku.com/heroku-x64.exe)
+      - [32-bit installer](https://cli-assets.heroku.com/heroku-x86.exe)
+        :::
+
+      ::::
+
+2. Login to Heroku from your CLI, following the command-line instructions:
 
-<code-block title="NPM">
-```sh
-npm install pg --save
+
+    ```bash
+    heroku login
+    ```
+
+
+### Create a Heroku project
+
+Create a new Heroku project by running the following command in the root directory of your Strapi project:
+
+```bash
+# path: ./my-project/
+heroku create
 ```
-</code-block>
 
-<code-block title="YARN">
-```sh
-yarn add pg
+:::tip
+You can use `heroku create custom-project-name`, to have Heroku create a `custom-project-name.heroku.com` URL. Otherwise, Heroku will automatically generate a random project name (and URL) for you.
+:::
+
+If you have a Heroku project app already created, you can use the following command to initialize your local project folder:
+
+```bash
+# path: ./my-project/
+heroku git:remote -a your-heroku-app-name
 ```
-</code-block>
 
-</code-group>
+Your local development environment is now set-up and configured to work with Heroku.
 
-:::::
+### Create a Heroku Database
 
-::::::
+The following command creates and connects a PostgreSQL database with your project. Consult the[ Heroku documentation](https://devcenter.heroku.com/articles/heroku-postgresql) for database plan names and costs.
 
-### 7. Commit your changes
+```bash
+#Path: ./my-project/
+heroku addons:create heroku-postgresql:<PLAN_NAME>
+```
 
-`Path: ./my-project/`
+You can retrieve the database credentials, which are stored as a string, with the config var name `DATABASE_URL` by running the following command in your terminal:
 
 ```bash
-git add .
-git commit -m "Update database config"
+# path: ./my-project/
+heroku config
 ```
 
-### 8. Update Yarn lockfile
+The command output has the form `DATABASE_URL: postgres://ebitxebvixeeqd:dc59b16dedb3a1eef84d4999sb4baf@ec2-50-37-231-192.compute-2.amazonaws.com: 5432/d516fp1u21ph7b`. The string has the structure `postgres://USERNAME:PASSWORD@HOST:PORT/DATABASE_NAME`
 
-`Path: ./my-project/`
+### Populate the environment variables
+
+You need to set the environment variables in Heroku for the database, server url, and secrets. The following commands set each variable:
```

Comment:
Can we specify these will pull these vars from your existing local `.env` file and to be careful then direct them to the openssl option you mention below as a best practice? (It's still good to keep the options but just need to add a warning)

Reply:
@derrickmehaffy I added a tabs card and reworked the intro so that setting new secrets is the first thing a user will see and it is listed a best practice. I think this works to address your comment but let me know if you think there is a better way to do this.